src/pages/RFOX/components/Claim/ClaimConfirm.tsx

```diff
@@ -37,23 +58,157 @@ const CustomRow: React.FC<RowProps> = props => <Row fontSize='sm' fontWeight='me
+
+  const claimAssetMarketDataUserCurrency = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, stakingAsset?.assetId ?? ''),
+  )
+
+  const stakingAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(claimQuote.claimAmountCryptoBaseUnit, stakingAsset?.precision ?? 0),
+    [claimQuote.claimAmountCryptoBaseUnit, stakingAsset?.precision],
+  )
+
+  const stakingAssetAccountAddress = useMemo(
+    () => fromAccountId(claimQuote.claimAssetAccountId).account,
+    [claimQuote.claimAssetAccountId],
+  )
+
+  const stakingAssetAccountNumberFilter = useMemo(() => {
+    return {
+      assetId: claimQuote.claimAssetId,
+      accountId: claimQuote.claimAssetAccountId,
+    }
+  }, [claimQuote.claimAssetAccountId, claimQuote.claimAssetId])
+
+  const stakingAssetAccountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, stakingAssetAccountNumberFilter),
   )
 
+  const claimAmountUserCurrency = useMemo(
+    () =>
+      bnOrZero(stakingAmountCryptoPrecision)
+        .times(claimAssetMarketDataUserCurrency.price)
+        .toFixed(),
+    [claimAssetMarketDataUserCurrency.price, stakingAmountCryptoPrecision],
+  )
+
+  const callData = useMemo(() => {
+    if (!stakingAsset) return
+
+    return encodeFunctionData({
+      abi: foxStakingV1Abi,
+      functionName: 'withdraw',
+      args: [BigInt(claimQuote.index)],
+    })
+  }, [stakingAsset, claimQuote.index])
+
+  const {
+    mutateAsync: handleClaim,
+    isIdle: isClaimMutationIdle,
+    isPending: isClaimMutationPending,
+    isSuccess: isClaimMutationSuccess,
+  } = useMutation({
+    mutationFn: async () => {
+      if (!wallet || stakingAssetAccountNumber === undefined || !stakingAsset || !callData) return
+
+      const adapter = assertGetEvmChainAdapter(stakingAsset.chainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: stakingAssetAccountNumber,
+        adapter,
+        data: callData,
+        value: '0',
+        to: RFOX_PROXY_CONTRACT_ADDRESS,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txId: string | undefined) => {
+      if (!txId) return
+
+      setClaimTxid(txId)
+    },
+  })
+
+  const isGetClaimFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        isClaimMutationIdle &&
+          stakingAssetAccountNumber !== undefined &&
+          wallet &&
+          stakingAsset &&
+          callData &&
+          feeAsset &&
+          feeAssetMarketDataUserCurrency,
+      ),
+    [
+      isClaimMutationIdle,
+      stakingAssetAccountNumber,
+      wallet,
+      stakingAsset,
+      callData,
+      feeAsset,
+      feeAssetMarketDataUserCurrency,
+    ],
+  )
+
+  const {
```

Comment:
@0xApotheosis doesn't have to be tackled in this PR since it will most likely go in first, but #7079 brings this guy which we will want to use
https://github.com/shapeshift/web/blob/98a2641927ddcbb9f2f6b96dc6874d0dc5115257/src/hooks/queries/useEvmFees.ts#L32